File: helpdesk-pro/frontend/src/app/core/services/api-key.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../models';

export interface ApiKey {
  id: string;
  name: string;
  keyPrefix: string;
  createdBy: string;
  lastUsedAt: string | null;
  expiresAt: string | null;
  active: boolean;
  createdAt: string;
}

export interface GeneratedApiKey {
  id: string;
  name: string;
  key: string;
  prefix: string;
  createdAt: string;
}

@Injectable({ providedIn: 'root' })
export class ApiKeyService {
  private base = `${environment.apiUrl}/api-keys`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<ApiKey[]> {
    return this.http.get<ApiResponse<ApiKey[]>>(this.base).pipe(map(r => r.data));
  }

  generate(name: string, expiresAt?: string): Observable<GeneratedApiKey> {
    return this.http.post<ApiResponse<GeneratedApiKey>>(this.base, { name, expiresAt }).pipe(map(r => r.data));
  }

  revoke(id: string): Observable<void> {
    return this.http.delete<ApiResponse<void>>(`${this.base}/${id}`).pipe(map(() => undefined));
  }
}

File: helpdesk-mobile/lib/core/services/api_key_service.dart
import '../api/api_client.dart';
import '../api/api_endpoints.dart';
import '../models/api_key_model.dart';

class ApiKeyService {
  final ApiClient _api;
  ApiKeyService(this._api);

  Future<List<ApiKeyModel>> getAll() async {
    final resp = await _api.get(ApiEndpoints.apiKeys);
    final data = resp.data['data'];
    final list = data is List ? data : (data['content'] ?? []);
    return List<ApiKeyModel>.from(list.map((e) => ApiKeyModel.fromJson(e)));
  }

  Future<Map<String, dynamic>> create(Map<String, dynamic> data) async {
    final resp = await _api.post(ApiEndpoints.apiKeys, data: data);
    return Map<String, dynamic>.from(resp.data['data']);
  }

  Future<void> revoke(String id) async {
    await _api.delete('${ApiEndpoints.apiKeys}/$id');
  }
}
